File: model/src/main/java/no/nav/pensjon/vtp/testmodell/repo/TestscenarioRepository.java
package no.nav.pensjon.vtp.testmodell.repo;

import java.util.Optional;
import java.util.stream.Stream;

public interface TestscenarioRepository {
    Stream<Testscenario> findAll();

    Optional<Testscenario> findById(String id);

    void delete(String id);

    Testscenario save(Testscenario testscenario);
}

File: model/src/main/java/no/nav/pensjon/vtp/testmodell/repo/TestscenarioRepository.kt
package no.nav.pensjon.vtp.testmodell.repo

import org.springframework.data.repository.Repository
import java.util.*
import java.util.stream.Stream

interface TestscenarioRepository : Repository<Testscenario, String> {
    fun findAll(): List<Testscenario>
    fun findById(id: String): Testscenario?
    fun deleteById(id: String)
    fun save(testscenario: Testscenario): Testscenario
}
